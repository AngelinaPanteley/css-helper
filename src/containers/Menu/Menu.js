import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import styles from './Menu.scss';
import * as actions from '../../store/actions/index';
import Icon from '../../components/UI/Icon/Icon';
import Dropdown from '../../components/UI/Dropdown/Dropdown';

class Menu extends Component {
  state = {
    isDropdownOpen: false,
  }

  onMenuClick = (e) => {
    e.preventDefault();
    this.toggleDropdown();
  }

  toggleDropdown = () => {
    this.setState(prevState => {
      return {
        isDropdownOpen: !prevState.isDropdownOpen
      }
    });
  }

  render() {
    const links = this.props.editorNames.map((elem) => {
      return {
        title: 'Create ' + elem[0].toUpperCase() + elem.slice(1),
        route: '/' + elem,
      }
    });

    return (
      <div className={styles.Menu}>
        <a href="" onClick={this.onMenuClick}>
          <Icon icon='menu' />
        </a>
        {
          <Dropdown isOpen={this.state.isDropdownOpen}
            links={links} left
            onClick={this.props.turnEditingModeOff}
            onClose={this.toggleDropdown} />
        }
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    editorNames: state.editors.names,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    turnEditingModeOff: () => dispatch(actions.turnEditingModeOff()),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Menu);