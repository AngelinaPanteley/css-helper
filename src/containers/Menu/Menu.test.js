"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import { MemoryRouter } from "react-router-dom";
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import Menu from './Menu.js';
import reducer from '../../store/reducers/index';

configure({ adapter: new Adapter() });
const composeEnhancers = process.env.NODE_ENV === 'development' ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ : null || compose;

const store = createStore(reducer, composeEnhancers(
  applyMiddleware(thunk)
));

describe('>>>Menu', () => {

  const menuComponent = <Provider store={store}>
    <MemoryRouter>
      <Menu />
    </MemoryRouter>
  </Provider>;

  const component = renderer.create(menuComponent);

  it('renders itself correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });
});
