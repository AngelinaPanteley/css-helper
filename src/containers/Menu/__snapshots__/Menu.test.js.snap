// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Menu renders itself correctly 1`] = `
<div>
  <a
    href=""
    onClick={[Function]}
  >
    <i
      className="undefined undefined"
    />
  </a>
  <div
    className="  "
  >
    <ul />
  </div>
</div>
`;
